jvm/src/main/scala/uk/gov/homeoffice/drt/db/dao/StaffShiftsDao.scala

```diff
@@ -27,11 +34,33 @@ case class StaffShiftsDao(db: CentralDatabase) extends IStaffShiftsDao {
   override def insertOrUpdate(staffShiftRow: StaffShiftRow): Future[Int] =
     db.run(staffShiftsTable.insertOrUpdate(staffShiftRow))
 
+  def updateStaffShift(previousStaffShiftRow: StaffShiftRow, staffShiftRow: StaffShiftRow): Future[Int] = {
+    val deleteAction = staffShiftsTable
+      .filter(row =>
+        row.port === previousStaffShiftRow.port &&
+          row.terminal === previousStaffShiftRow.terminal &&
+          row.shiftName === previousStaffShiftRow.shiftName &&
+          row.startDate === previousStaffShiftRow.startDate &&
+          row.startTime === previousStaffShiftRow.startTime
+      ).delete
+
+    val updatesStaffShiftRow = if (previousStaffShiftRow.endDate.isDefined) staffShiftRow.copy(endDate = previousStaffShiftRow.endDate) else staffShiftRow.copy(endDate = None)
+
+    val insertAction = staffShiftsTable += updatesStaffShiftRow
+
+    if (staffShiftRow.startDate.getTime > new Date(System.currentTimeMillis()).getTime && previousStaffShiftRow.startDate.getTime < new Date(System.currentTimeMillis()).getTime && previousStaffShiftRow.endDate.isEmpty) {
```

Comment:
We have logic here to behave differently according to start and end times. We're missing tests around it to enforce expected behaviour